frontend/components/node-item/node-item.ts

```diff
@@ -33,10 +33,9 @@ export class NodeItem {
 
     // Listen for changes to selected node
     this.componentDataStore.dataStream
-      .pluck('selectedNode')
-      .distinctUntilChanged((selectedNode: any) => {
-        return selectedNode ? selectedNode.id : '';
-      })
+      .map(({ selectedNode }: any) => selectedNode)
+      .filter((selectedNode: any) => selectedNode && selectedNode.id)
+      .distinctUntilKeyChanged('id')
```

Comment:
It could be that it is part of alpha6 for @reactivex and might be part of the newer typings, if thats the case I think there is no harm in upgrading to version 6 in our package.json